### PR TITLE
Issue 356: Release notes 4.5.0

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

- summary for release 4.5.0
- highlights for 4.5.0
- full list of JIRA and Github issues.

### CODE REVIEW

**Comment by sijie** [comment]
https://sijie.github.io/bookkeeper-staging-site/docs/latest/releaseNotes/ for easy to review

**Comment by jvrao** [comment]
+1

**Comment by eolivelli** [comment]
+1 look great to me.
I don't remember that in 4.4 it was possible to use digest md5 auth

File: site/docs/latest/releaseNotes.md [hunk 1]
```diff
@@ -0,0 +1,483 @@
+---
```
Comment (jvrao) [inline site/docs/latest/releaseNotes.md]: Can we add all upgrades we did?

Upgrade protobuf to 2.6
Codehale to 3.1.0
ZK to version 3.5
Move to JDK 8
SLF4J to 7.25
Netty 4.1
Guava to version 20.0

**Comment by eolivelli** [comment]
+1 look great to me.
I don't remember that in 4.4 it was possible to use digest md5 auth.
About guava we should tell that we are shading it and it is used only internally, users will be able to use any version of that library

File: site/docs/latest/releaseNotes.md [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+Prior to 4.5.0, the `LAC` is only advanced when subsequent entries are added. If there is no subsequent entries added,
+the last entry written will not be visible to readers until the ledger is closed. High-level client (e.g. DistributedLog) or applications
+has to work around this by writing some sort of `control records` to advance `LAC`.
+
+In 4.5.0, a new `explicit lac` feature is introduced to periodically advance `LAC` if there are not subsequent entries added. This feature
+can be enabled by setting `explicitLacInterval` to a positive value.
+
+### Performance
+
+There are a lot for performance related bug fixes and improvements in 4.5.0. The major performance improvement introduced in 4.5.0, is
```
Comment (jvrao) [inline site/docs/latest/releaseNotes.md]: I think there are lot more perf improvements related to  caching additional metadata, avoiding buffer copies, taking advantage of java8 features, scheduling and threading improvements etc. Ledger recovery improvements. We should discuss these.

File: site/docs/latest/releaseNotes.md [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+
+#### Customized Ledger Metadata
+
+A `Map<String, byte[]>` is introduced in ledger metadata in 4.5.0. Clients now are allowed to pass in a key/value map when creating ledgers.
+This customized ledger metadata can be later on used by user defined placement policy. This extends the flexibility of bookkeeper API.
+
+#### Add Prometheus stats provider
+
+A new [Prometheus](https://prometheus.io/) [stats provider](https://github.com/apache/bookkeeper/tree/master/bookkeeper-stats-providers/prometheus-metrics-provider)
+is introduce in 4.5.0. It simplies the metric collection when running bookkeeper on [kubernetes](https://kubernetes.io/).
```
Comment (jvrao) [inline site/docs/latest/releaseNotes.md]: We made ton of test improvements. We need to mention that too.

Comment (jvrao) [inline site/docs/latest/releaseNotes.md]: We may need to talk about tool improvements too?

File: site/docs/latest/releaseNotes.md (hunk 1 above)
Comment (sijie) [inline site/docs/latest/releaseNotes.md]: done

File: site/docs/latest/releaseNotes.md (hunk 3 above)
Comment (sijie) [inline site/docs/latest/releaseNotes.md]: I don't think test improvements need to be in `highlights`.

I will include the tool improvements.

**Comment by sijie** [comment]
@eolivelli we shaded guava since 4.4. it is not actually new to 4.5. I would prefer we documenting somewhere, rather than including it in 4.5 release notes. The 4.5 release notes should focus on the changes in 4.5, it is not a documentation.

File: site/docs/latest/releaseNotes.md (hunk 2 above)
Comment (sijie) [inline site/docs/latest/releaseNotes.md]: I added a paragraph at the beginning of `performance` section.

**Comment by sijie** [comment]
addressed the comments. please take a look again.

**Comment by jvrao** [comment]
@sijie  with these changes it looks good.

**Comment by eolivelli** [comment]
I forgot to mention networkless bookie but I think it does not make sense yet for the general audience

**Comment by eolivelli** [comment]
I forgot to mention networkless bookie but I think it does not make sense yet for the general audience

**Comment by sijie** [comment]
update the release notes

**Comment by sijie** [comment]
release notes are ready to merge now.

**Pull request closed without merging**